modules/twinkleprotect.js

```diff
@@ -240,13 +242,28 @@ Twinkle.protect.callback.showLogAndCurrentProtectInfo = function twinkleprotectC
 
 		if (Twinkle.protect.hasProtectLog) {
 			$linkMarkup.append(
-				$('<a target="_blank" href="' + mw.util.getUrl('Special:Log', {action: 'view', page: mw.config.get('wgPageName'), type: 'protect'}) + '">protection log</a>'),
-				Twinkle.protect.hasStableLog ? $('<span> &bull; </span>') : null
-			);
+				$('<a target="_blank" href="' + mw.util.getUrl('Special:Log', {action: 'view', page: mw.config.get('wgPageName'), type: 'protect'}) + '">protection log</a>'));
+			if (!currentlyProtected || (!Twinkle.protect.currentProtectionLevels.edit && !Twinkle.protect.currentProtectionLevels.move)) {
+				var lastProtectAction = Twinkle.protect.protectLog[0];
+				if (lastProtectAction.action === 'unprotect') {
+					$linkMarkup.append(' (unprotected ' + new Morebits.date(lastProtectAction.timestamp).calendar('utc') + ')');
+				} else { // protect or modify
+					$linkMarkup.append(' (expired ' + new Morebits.date(lastProtectAction.params.details[0].expiry).calendar('utc') + ')');
+				}
+			}
+			$linkMarkup.append(Twinkle.protect.hasStableLog ? $('<span> &bull; </span>') : null);
 		}
 
 		if (Twinkle.protect.hasStableLog) {
 			$linkMarkup.append($('<a target="_blank" href="' + mw.util.getUrl('Special:Log', {action: 'view', page: mw.config.get('wgPageName'), type: 'stable'}) + '">pending changes log</a>)'));
+			if (!currentlyProtected || !Twinkle.protect.currentProtectionLevels.stabilize) {
+				var lastStabilizeAction = Twinkle.protect.stableLog[0];
+				if (lastStabilizeAction.action === 'reset') {
+					$linkMarkup.append(' (reset ' + new Morebits.date(lastStabilizeAction.timestamp).calendar('utc') + ')');
+				} else { // config or modify
+					$linkMarkup.append(' (expired ' + new Morebits.date(lastStabilizeAction.params.expiry).calendar('utc') + ')');
```

Comment:
[Right here](https://phabricator.wikimedia.org/T267922)

Reply:
It's a [standard MW timestamp format](https://www.mediawiki.org/wiki/Manual:Timestamp) – comes up in a lot of places. Should we [tweak our constructor to take care of it](https://github.com/siddharthvp/mwn/blob/master/src/date.ts#L21)?

Reply:
Good point, so I guess so?  I consider it a bug as far as the return is considered, but I suppose it'd only help.  It's used in URLs, at least.

Reply:
Dunno if you were gonna submit it, but #1200